templates/unnest_parquet_view.sql.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+"""Generate view to unnest parquet-format list and map fields."""
+
+from argparse import ArgumentParser
+from textwrap import dedent
+import json
+import subprocess
+import sys
+
+
+def qualify(table, dataset, project):
+    """Fully qualify table in standard sql format without `s."""
+    table = table.replace("`", "").replace(":", ".")
+    dots = table.count(".")
+    if dots == 0:
+        return f"{project}.{dataset}.{table}"
+    elif dots == 1:
+        return f"{project}.{table}"
+    else:
+        return table
+
+
+def replace(field, unnest_layer=0, *prefix):
+    """Convert BigQuery field to a SQL expression for use in a REPLACE block."""
+    if field["type"] == "RECORD":
+        if len(field["fields"]) == 1:
+            # prevent naming collisions via `UNNEST(...) AS`
```

Comment:
Was there a particular case that prompted this? It seems to me the engine would do the right thing by grabbing the "closest" column

Reply:
i confirmed in testing that it grabs the "farthest" column